In [1]:
import pandas as pd
import numpy as np

## Questions:
1. Obviosly the most relevant question will be related to knowing what are the key factors that contributes to booking a listing
2. Then does this hypthesis hold to all states, here will will investigate Seattle and Boston whom I believe are on 2 different coasts and might be different
3. 

In [58]:
# seattle_cal = pd.read_csv('Data/AirBnB/seattle/calendar.csv')
seattle_list = pd.read_csv('Data/AirBnB/seattle/listings.csv')
# seattle_rev = pd.read_csv('Data/AirBnB/seattle/reviews.csv')

In [14]:
# Context manager to temporarily set options in the with statement context
def display_all(df):
    with pd.option_context('display.max_rows', 1000, 'display.max_columns', 1000):
        display(df)

In [16]:
display_all(seattle_list.head())

id                           listing_url       scrape_id last_scraped  \
0   241032   https://www.airbnb.com/rooms/241032  20160104002432   2016-01-04   
1   953595   https://www.airbnb.com/rooms/953595  20160104002432   2016-01-04   
2  3308979  https://www.airbnb.com/rooms/3308979  20160104002432   2016-01-04   
3  7421966  https://www.airbnb.com/rooms/7421966  20160104002432   2016-01-04   
4   278830   https://www.airbnb.com/rooms/278830  20160104002432   2016-01-04   

                                  name  \
0         Stylish Queen Anne Apartment   
1   Bright & Airy Queen Anne Apartment   
2  New Modern House-Amazing water view   
3                   Queen Anne Chateau   
4       Charming craftsman 3 bdm house   

                                             summary  \
0                                                NaN   
1  Chemically sensitive? We've removed the irrita...   
2  New modern house built in 2013.  Spectacular s...   
3  A charming apartment that sits atop Queen Anne...   
4  Cozy family craftman house in beautiful neighb...   

                                               space  \
0  Make your self at home in this charming one-be...   
1  Beautiful, hypoallergenic apartment in an extr...   
2  Our house is modern, light and fresh with a wa...   
3                                                NaN   
4  Cozy family craftman house in beautiful neighb...   

                                         description experiences_offered  \
0  Make your self at home in this charming one-be...                none   
1  Chemically sensitive? We've removed the irrita...                none   
2  New modern house built in 2013.  Spectacular s...                none   
3  A charming apartment that sits atop Queen Anne...                none   
4  Cozy family craftman house in beautiful neighb...                none   

                               neighborhood_overview  \
0                                                NaN   
1  Queen Anne is a wonderful, truly functional vi...   
2  Upper Queen Anne is a charming neighborhood fu...   
3                                                NaN   
4  We are in the beautiful neighborhood of Queen ...   

                                               notes  \
0                                                NaN   
1  What's up with the free pillows?  Our home was...   
2  Our house is located just 5 short blocks to To...   
3                                                NaN   
4                                          Belltown    

                                             transit  \
0                                                NaN   
1  Convenient bus stops are just down the block, ...   
2  A bus stop is just 2 blocks away.   Easy bus a...   
3                                                NaN   
4  The nearest public transit bus (D Line) is 2 b...   

                                       thumbnail_url  \
0                                                NaN   
1  https://a0.muscache.com/ac/pictures/14409893/f...   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                          medium_url  \
0                                                NaN   
1  https://a0.muscache.com/im/pictures/14409893/f...   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                         picture_url  \
0  https://a1.muscache.com/ac/pictures/67560560/c...   
1  https://a0.muscache.com/ac/pictures/14409893/f...   
2  https://a2.muscache.com/ac/pictures/b4324e0f-a...   
3  https://a0.muscache.com/ac/pictures/94146944/6...   
4  https://a1.muscache.com/ac/pictures/6120468/b0...   

                                      xl_picture_url   host_id  \
0                                                

In [12]:
seattle_list.shape

(3818, 92)

In [59]:
seattle_list.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'name', 'summary',
       'space', 'description', 'experiences_offered', 'neighborhood_overview',
       'notes', 'transit', 'thumbnail_url', 'medium_url', 'picture_url',
       'xl_picture_url', 'host_id', 'host_url', 'host_name', 'host_since',
       'host_location', 'host_about', 'host_response_time',
       'host_response_rate', 'host_acceptance_rate', 'host_is_superhost',
       'host_thumbnail_url', 'host_picture_url', 'host_neighbourhood',
       'host_listings_count', 'host_total_listings_count',
       'host_verifications', 'host_has_profile_pic', 'host_identity_verified',
       'street', 'neighbourhood', 'neighbourhood_cleansed',
       'neighbourhood_group_cleansed', 'city', 'state', 'zipcode', 'market',
       'smart_location', 'country_code', 'country', 'latitude', 'longitude',
       'is_location_exact', 'property_type', 'room_type', 'accommodates',
       'bathrooms', 'bedrooms', 'beds', 'bed_type', 'amenities', '

Looking at the listing file, we see 92 variables with 3,818 records. Many of these variables makes sense but several variables are not clear which I will explore below

In [60]:
# Scrape_id and last_scraped
seattle_list.scrape_id.value_counts(), seattle_list.last_scraped.value_counts()

(20160104002432    3818
 Name: scrape_id, dtype: int64, 2016-01-04    3818
 Name: last_scraped, dtype: int64)

It seems that these are the extraction data/time of the data set available on hand and can be easily dropped from the dataset

In [61]:
df_list = seattle_list.drop(['scrape_id', 'last_scraped'], axis=1)

Now there are variables that simply add no value to the analysis we are about to do

In [66]:
# listing_url
df_list.listing_url.nunique()

3818

It seems that every lisitng has its unique URL (totally expected) and can be dropped from the dataframe

In [67]:
df_list.drop('listing_url',axis=1, inplace=True)

I was thinking that the names shall be unique because no one wants to have his or her listing with the same of a competitor especially when they are in the same state and maybe city because a bad review or word of mouth can be harmful

> We hve 51 listings with the same name

In [100]:
# We have 51 listings with the same name
df_list.name.value_counts()[:25].sum()

51

Is it an issue in the extration or do 2 different listing share the same name?

In [116]:
df_list.name.value_counts()[:25]

Capitol Hill Apartment                 3
Downtown Seattle Pike Place Market     2
Capitol Hill Studio Apartment          2
Private bath walk UW U Village(202)    2
Fabulous Seattle Modern                2
"THE 5-STAR HOUSE"                     2
Sans souci in Seattle                  2
Charming Craftsman in Ballard          2
A Quiet Room in a House by the Lake    2
Cozy Ballard Apartment                 2
Cozy studio in Capitol Hill!           2
comfy room&convenient location（中英）     2
Ravenna Retreat                        2
Secluded Setting in North Seattle      2
2 Baths W/D,View walk UW (302)         2
Center UW Blast Internet New #102      2
Privacy & gorgeous panoramic views!    2
Capitol Hill Master Bedroom            2
Vintage Apt in Heart of CapitolHill    2
In-City Suite Retreat                  2
Seattle Downtown Convention Center     2
Downtown studio close to everything    2
Private room in West Seattle           2
Bedroom in artistic Northwest home     2
Luxury Apt/Downt

In [121]:
df_list.groupby(['name', 'id']).id.count().sort_values(ascending=False).head()

name                             id     
• West Seattle Crash Pad •       8102349    1
Cozy Room West Seattle           6888107    1
Cozy Seattle Bungalow            6729818    1
Cozy Seattle Bedroom             9823290    1
Cozy Seattle Apartment Near UW   202260     1
Name: id, dtype: int64

In [122]:
display_all(df_list[df_list.name == 'comfy room&convenient location（中英）'])

id                                name  \
1958  9814376  comfy room&convenient location（中英）   
2052  9863457  comfy room&convenient location（中英）   

                                                summary  \
1958  clean and comfy bedroom&bathroom,located in to...   
2052  clean and comfy bedroom&bathroom,located in to...   

                                                  space  \
1958  Big,quiet and clean room in lovely townhouse,r...   
2052  Big,quiet and clean room in lovely townhouse,r...   

                                            description experiences_offered  \
1958  clean and comfy bedroom&bathroom,located in to...                none   
2052  clean and comfy bedroom&bathroom,located in to...                none   

                                  neighborhood_overview  \
1958  Located in townhouse and features a wide array...   
2052  Located in townhouse and features a wide array...   

                                                  notes  \
1958  For some reasons,the owner is not sleep in the...   
2052  The guest can use the whole building,including...   

                                                transit  \
1958  clean and comfy bedroom&bathroom,located in to...   
2052  clean and comfy bedroom&bathroom,located in to...   

                                          thumbnail_url  \
1958  https://a2.muscache.com/ac/pictures/fbe453b7-e...   
2052  https://a2.muscache.com/ac/pictures/febb7afd-d...   

                                             medium_url  \
1958  https://a2.muscache.com/im/pictures/fbe453b7-e...   
2052  https://a2.muscache.com/im/pictures/febb7afd-d...   

                                            picture_url  \
1958  https://a2.muscache.com/ac/pictures/fbe453b7-e...   
2052  https://a2.muscache.com/ac/pictures/febb7afd-d...   

                                         xl_picture_url   host_id  \
1958  https://a2.muscache.com/ac/pictures/fbe453b7-e...  50557855   
2052  https://a2.muscache.com/ac/pictures/febb7afd-d...  50557855   

                                        host_url host_name  host_since  \
1958  https://www.airbnb.com/users/show/50557855       Hua  2015-12-05   
2052  https://www.airbnb.com/users/show/50557855       Hua  2015-12-05   

     host_location host_about host_response_time host_response_rate  \
1958            US        NaN       within a day               100%   
2052            US        NaN       within a day               100%   

     host_acceptance_rate host_is_superhost  \
1958                  NaN                 f   
2052                  NaN                 f   

                                     host_thumbnail_url  \
1958  https://a2.muscache.com/ac/pictures/8f167159-e...   
2052  https://a2.muscache.com/ac/pictures/8f167159-e...   

                                       host_picture_url host_neighbourhood  \
1958  https://a2.muscache.com/ac/pictures/8f167159-e...         Holly Park   
2052  https://a2.muscache.com/ac/pictures/8f167159-e...         Holly Park   

      host_listings_count  host_total_listings_count  \
1958                  2.0                        2.0   
2052                  2.0                        2.0   

                         host_verifications host_has_profile_pic  \
1958  ['email', 'phone', 'google', 'jumio']                    t   
2052  ['email', 'phone', 'google', 'jumio']                    t   

     host_identity_verified  \
1958                      t   
2052                      t   

                                                 street neighbourhood  \
1958  South Holly Park Drive, Seattle, WA 98118, Uni...    Holly Park   
2052  South Holly Park Drive, Seattle, WA 98118, Uni...    Holly Park   

     neighbourhood_cleansed neighbourhood_group_cleansed     city state  \
1958      South Beacon Hill                  Beacon Hill  Seattle    WA   
2052             Holly Park                  Beacon Hill  Seattle    WA   

     zipcode   market smart_location country_code        country   latitude

Although after manual inspection of these properties, it seems that these listings are not duplicates due to an error in data extarction but it is clear that although they are very similar, they are 2 distinct listings and can't be dropped, but for analysis purposes I believe the name holds little value and can be dropped especially that it is mostly a combination of the apartment or room location and some features 

In [126]:
df_list.drop('name', axis=1, inplace=True)

The variables 
- summary
- spac
- description
- neighborhood_overview
- transit
- notes 

although full of text I believe holds a great value in assisting a traveler in selecting a listing and in helping me answer questiosn posted above

Drop experiences_offered as it hold no value

In [145]:
df_list.experiences_offered.value_counts()

none    3818
Name: experiences_offered, dtype: int64

In [146]:
df_list.drop('experiences_offered', axis=1, inplace=True)

Drop all url variables as they hold no value to the analysis

In [156]:
col_url = df_list.columns[df_list.columns.str.contains('_url')]
df_list.drop(col_url, axis=1, inplace=True)

In [165]:
df_list.columns[df_list.isnull().sum()/len(df_list) > .75]

Index(['square_feet', 'license'], dtype='object')

In [166]:
df_list.drop(['square_feet', 'license'], axis=1,inplace=True)

It seems that the dataframe now ha

In [168]:
df_list.shape

(3818, 78)

In [167]:
display_all(df_list.tail(10))

id                                            summary  \
3808   1844791  Our charming home in fabulous Upper Fremont is...   
3809   6120046  From the deck of this quaint little apartment ...   
3810    262764  Our 2BR/1 bath home in Fremont's most fun neig...   
3811   8578490  In the true spirit of AirBNB this unit is avai...   
3812   3383329  Of the Tree & Clouds' "Roots" apartment is 8 b...   
3813   8101950  Our 3BR/2BA house boasts incredible views of t...   
3814   8902327  800 square foot 1 bedroom basement apartment w...   
3815  10267360  Very comfortable lower unit. Quiet, charming m...   
3816   9604740  Cozy studio condo in the heart on Madison Park...   
3817  10208623  All hardwood floors, fireplace, 65" TV with Xb...   

                                                  space  \
3808  This is a classically beautiful Craftsman home...   
3809  This one bedroom apartment (Hippy Shack) is de...   
3810  1200 ft2 2BR, 1 Bath farmhouse with a large, t...   
3811  The building is older (1970's) so not the mode...   
3812  The 1-bedroom, ground-floor Roots apartment is...   
3813  Our 3BR/2BA house bright, stylish, and wheelch...   
3814  This space has a great view of Portage Bay wit...   
3815                                                NaN   
3816  Fully furnished unit to accommodate most needs...   
3817                                                NaN   

                                            description  \
3808  Our charming home in fabulous Upper Fremont is...   
3809  From the deck of this quaint little apartment ...   
3810  Our 2BR/1 bath home in Fremont's most fun neig...   
3811  In the true spirit of AirBNB this unit is avai...   
3812  Of the Tree & Clouds' "Roots" apartment is 8 b...   
3813  Our 3BR/2BA house boasts incredible views of t...   
3814  800 square foot 1 bedroom basement apartment w...   
3815  Very comfortable lower unit. Quiet, charming m...   
3816  Cozy studio condo in the heart on Madison Park...   
3817  All hardwood floors, fireplace, 65" TV with Xb...   

                                  neighborhood_overview  \
3808  Fremont is wonderful; you will love it here. T...   
3809  This cottage apartment is located just a few b...   
3810  We love that Fremont is so centrally located t...   
3811  I am equidistant to Fremont and Wallingford an...   
3812  It's taken me a while to figure out why Fremon...   
3813  We're located near lots of family fun. Woodlan...   
3814  The neighborhood is a quiet oasis that is clos...   
3815                                                NaN   
3816  Madison Park offers a peaceful slow pace upsca...   
3817                                                NaN   

                                                  notes  \
3808  Normal check-in time is 3-4:00; whenever possi...   
3809  1. Is the property quite private? ie. can we s...   
3810  We have two cats, Polly and Theo. They are ind...   
3811  Laundry is available but is $1.75 to wash and ...   
3812  There is NO television. We do have good WI-FI....   
3813                                                NaN   
3814  This is a basement apartment in a newer reside...   
3815                                                NaN   
3816                                                NaN   
3817               Also our puppy will be boarded away.   

                                                transit   host_id  \
3808  The #5 bus is 2 blocks away and it is a 15-20 ...   7964921   
3809  This cottage apartment is located just a few b...  31752892   
3810  We're centrally located to several bus stops t...   1379138   
3811  How to Get to: UW/Husky Stadium):  10 minute d...  11847918   
3812  The apartment is easily accessed from I-5 Nort...   3443346   
3813                                                NaN  31148752   
3814  Uber and Car2go are good options in Seattle. T...  46566046   
3815                                                NaN  52791370   
3816                           